File: Ejs16 PreParcial/craps.py
import random
import time
import math
def ej1():
    random.seed(time.time())
    dice1 = random.randint(1, 6)
    dice2 = random.randint(1, 6)
    initialRoll = dice1 + dice2
    
    playerinput = int(input(('Bienvenido al juego Craps, escriba 1 para lanzar los dados: \n')))
    if playerinput == 1:
        if initialRoll == 7 or initialRoll == 11:
            print(f'El resultado de los dados fue {initialRoll}, ganaste')
        elif initialRoll == 2 or initialRoll == 3 or initialRoll == 12:
            print(f'El resultado de los dados fue {initialRoll}, perdiste')
        else:
            stop = False
            print(f"El resultado de los dados fue {initialRoll}")
            playerinput = int(input('Escriba 1 para lanzar los dados nuevamente: \n'))
            if playerinput == 1:
                    random.seed(time.time())
                    dice1 = random.randint(1, 6)
                    dice2 = random.randint(1, 6)
                    newRoll = dice1 + dice2
                    if newRoll == initialRoll:
                        print(f'El resultado de los dados fue {newRoll}, ganaste')
                        stop = True
                        
                    elif newRoll == 7:
                        print(f'El resultado de los dados fue {newRoll}, perdiste')
                        stop = True
                    else:
                        print(f'El resultado de los dados fue {newRoll}')
                        
            if not stop:            
                playerinput = int(input('Escriba 1 para lanzar los dados nuevamente: \n'))
                if playerinput == 1:
                        random.seed(time.time())
                        dice1 = random.randint(1, 6)
                        dice2 = random.randint(1, 6)
                        newRoll = dice1 + dice2
                        if newRoll == initialRoll:
                            print(f'El resultado de los dados fue {newRoll}, ganaste')
                            stop = True
                        elif newRoll == 7:
                            print(f'El resultado de los dados fue {newRoll}, perdiste')
                            stop = True
                        else:
                            print(f'El resultado de los dados fue {newRoll}')
                            
            
                if not stop:
                    playerinput = int(input('Escriba 1 para lanzar los dados nuevamente: \n'))
                    if playerinput == 1:
                            random.seed(time.time())
                            dice1 = random.randint(1, 6)
                            dice2 = random.randint(1, 6)
                            newRoll = dice1 + dice2
                            if newRoll == initialRoll:
                                print(f'El resultado de los dados fue {newRoll}, ganaste')
                                
                            elif newRoll == 7:
                                print(f'El resultado de los dados fue {newRoll}, perdiste')
                                
                            else:
                                print(f'El resultado de los dados fue {newRoll}')
                    if not stop:
                        playerinput = int(input('Escriba 1 para lanzar los dados nuevamente: \n'))
                        if playerinput == 1:
                                random.seed(time.time())
                                dice1 = random.randint(1, 6)
                                dice2 = random.randint(1, 6)
                                newRoll = dice1 + dice2
                                if newRoll == initialRoll:
                                    print(f'El resultado de los dados fue {newRoll}, ganaste')
                                    
                                elif newRoll == 7:
                                    print(f'El resultado de los dados fue {newRoll}, perdiste')
                                    
                                else:
                                    print(f'El resultado de los dados fue {newRoll}')
                                                   
    else:
        print('Escriba 1 para lanzar los dados')

def ej2():
    x = float(input("Ingrese el valor de x: \n"))
    y = float(input("Ingrese el valor de y: \n"))
    
    z = math.sqrt(x + (math.pow(y,2)/(3 + x))) - ((math.pow(x,2)/(math.pow(y,3) + x))- ((5) * (math.sqrt(3 * x))));
    print(f"El resultado de la ecuación es: {z}")
def main():
    playerinput = int(input("Elija el ejercicio que desea ejecutar 1 o 2 : \n"))
    if playerinput == 1:
        ej1()
    elif playerinput == 2:
        ej2()
    else:
        print("Ejercicio no encontrado")
                
                

if __name__ == "__main__":
    main()

File: Ejs16 PreParcial/craps.cpp
#include <iostream>
#include <math.h>
using namespace std;

void ej1()
{
    srand(time(0));

    int dice1 = rand() % 6 + 1;
    int dice2 = rand() % 6 + 1;

    int initialRoll = dice1 + dice2;

    int input;
    cout << "Bienvenido al juego Craps, escriba 1 para tirar los dados" << endl;
    cin >> input;

    if (input == 1)
    {
        cout << "El resultado de los dados es: " << initialRoll << endl;
        if (initialRoll == 7 || initialRoll == 11)
        {
            cout << "Ganaste, con un resutado de " << initialRoll << endl;
        }
        else if (initialRoll == 2 || initialRoll == 3 || initialRoll == 12)
        {
            cout << "Perdiste, con un resutado de " << initialRoll << endl;
        }
        else
        {
            bool stop = false;

            srand(time(0));
            dice1 = rand() % 6 + 1;
            dice2 = rand() % 6 + 1;
            cout << "Tire nuevamente los dados, escriba 1 para lanzarlos" << endl;
            cin >> input;

            if (input == 1)
            {
                int newRoll = dice1 + dice2;
                cout << "El resultado de los dados es: " << newRoll << endl;
                if (newRoll == initialRoll)
                {
                    cout << "Ganaste, con un resutado de " << newRoll << endl;
                    stop = true;
                }
                else if (newRoll == 7)
                {
                    cout << "Perdiste, con un resutado de " << newRoll << endl;
                    stop = true;
                }
            }
            else
            {
                cout << "El input no es valido" << endl;
            }
            if (!stop)
            {
                srand(time(0));
                dice1 = rand() % 6 + 1;
                dice2 = rand() % 6 + 1;
                cout << "Tire nuevamente los dados, escriba 1 para lanzarlos" << endl;
                cin >> input;

                if (input == 1)
                {
                    int newRoll = dice1 + dice2;
                    cout << "El resultado de los dados es: " << newRoll << endl;
                    if (newRoll == initialRoll)
                    {
                        cout << "Ganaste, con un resutado de " << newRoll << endl;
                        stop = true;
                    }
                    else if (newRoll == 7)
                    {
                        cout << "Perdiste, con un resutado de " << newRoll << endl;
                        stop = true;
                    }
                }
                else
                {
                    cout << "El input no es valido" << endl;
                }
                if (!stop)
                {
                    srand(time(0));
                    dice1 = rand() % 6 + 1;
                    dice2 = rand() % 6 + 1;
                    cout << "Tire nuevamente los dados, escriba 1 para lanzarlos" << endl;
                    cin >> input;

                    if (input == 1)
                    {
                        int newRoll = dice1 + dice2;
                        cout << "El resultado de los dados es: " << newRoll << endl;
                        if (newRoll == initialRoll)
                        {
                            cout << "Ganaste, con un resutado de " << newRoll << endl;
                            stop = true;
                        }
                        else if (newRoll == 7)
                        {
                            cout << "Perdiste, con un resutado de " << newRoll << endl;
                            stop = true;
                        }
                    }
                    else
                    {
                        cout << "El input no es valido" << endl;
                    }
                    if (!stop)
                    {
                        srand(time(0));
                        dice1 = rand() % 6 + 1;
                        dice2 = rand() % 6 + 1;
                        cout << "Tire nuevamente los dados, escriba 1 para lanzarlos" << endl;
                        cin >> input;

                        if (input == 1)
                        {
                            int newRoll = dice1 + dice2;
                            cout << "El resultado de los dados es: " << newRoll << endl;
                            if (newRoll == initialRoll)
                            {
                                cout << "Ganaste, con un resutado de " << newRoll << endl;
                                stop = true;
                            }
                            else if (newRoll == 7)
                            {
                                cout << "Perdiste, con un resutado de " << newRoll << endl;
                                stop = true;
                            }
                        }
                        else
                        {
                            cout << "El input no es valido" << endl;
                        }
                    }
                }
            }
        }
    }
    else
    {
        cout << "El input no es valido" << endl;
    }
}

void ej2()
{

    float x;
    float y;

    cout << "Ingrese el valor de x" << endl;
    cin >> x;

    cout << "Ingrese el valor de y" << endl;
    cin >> y;

    float z = sqrt(x + (pow(y, 2) / (3 + x))) - ((pow(x, 2) / (pow(y, 3) + x)) - ((5) * (sqrt(3 * x))));

    cout << "El resultado de la operacion es: " << z << endl;
}
int main()
{

    int input;
    cout << "Escriba 1 para ver el ejercicio 1, 2 para el ejercicio 2" << endl;
    cin >> input;

    if (input == 1)
    {
        ej1();
    }
    else if (input == 2)
    {
        ej2();
    }
    else
    {
        cout << "El input no es valido" << endl;
    }
}
